src/runtime/audience-action-flow.js

```diff
@@ -290,11 +291,14 @@ export class AudienceActionFlow {
    * @private
    */
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  handleSurveyDataTransferRequest_(request) {
+  async handleSurveyDataTransferRequest_(request) {
     // @TODO(justinchou): execute callback with setOnInterventionComplete
     // then check for success
-    const gaLoggingSuccess = this.logSurveyDataToGoogleAnalytics(request);
-    if (gaLoggingSuccess) {
+    const {onResult} = this.params_;
+    const loggingSuccess = onResult
+      ? await this.params_.onResult(request)
```

Comment:
```suggestion
      ? await onResult(request)
```

Reply:
We might want to gracefully support rejects/throws from the `onResult` callback

Reply:
We could `await onResult(request)` within a `try/catch` block, and if it fails, we could set `loggingSuccess` to false

Reply:
Good catch! +1 to error handling

Reply:
Moved this area into a new function `attemptSurveyDataTransfer` and wrapped `onResult` in a try/catch with a warning if it catches an exception.